File: testeDevJR.js
// teste dev em JavaScript
var anagramaAgrupado = function (palavras) {
    
    const grupoAnagrama = {};

    for (let palavra of palavras) {

        const palavraOrdenada = palavra.split("").sort().join("");

        if (!grupoAnagrama[palavraOrdenada]) grupoAnagrama[palavraOrdenada] = [];

        grupoAnagrama[palavraOrdenada].push(palavra);
    }

    return Object.values(grupoAnagrama);
};

console.log(anagramaAgrupado(["nata", "sorvete", "anta", "paralelepipedo", "tana"]))

File: testeDevJR.py
#teste dev em python com back pois em C# o codigo ficou muito extenso
def anagramaAgrupado(palavras):

    grupoAnagrama = {}

    for palavra in palavras:
        palavraOrdenada = "".join(sorted(palavra))

        if palavraOrdenada in grupoAnagrama:
            grupoAnagrama[palavraOrdenada].append(palavra)
        else:
            grupoAnagrama[palavraOrdenada] = [palavra]

    return list(grupoAnagrama.values())

palavras = ['nata', 'sorvete', 'anta',
            'paralelepipedo', 'tana', 'dopelelerapaip']

print(anagramaAgrupado(palavras))
